feat(link-list-section): Web component created

RaphaelAmadeu opened the pull request:
### Related Ticket(s)

#3376
### Description

Created the web-component version of `LinkListSection` following it's design and functional specs.

asudoh commented on packages/web-components/src/components/link-list/link-list.ts:
```diff
@@ -87,6 +87,10 @@ class DDSLinkList extends StableSelectorMixin(LitElement) {
     window.removeEventListener('resize', this._handleResize);
   }
 
+  updated() {
+    sameHeight(this.childItems, 'md');
+  }
+
```
Let's leave this not running for now - We are trying to determine what to do with `sameWidth()`/`sameHeight()` as a whole.

```suggestion
```

asudoh reviewed (commented):
LGTM basically, just one thing - Thanks @RaphaelAmadeu!

asudoh reviewed (approved):
Thanks @RaphaelAmadeu for the update!

jeffchew reviewed (changes requested):
Visually looks broken to me, can you take a look @RaphaelAmadeu ?

<img width="993" alt="Screen Shot 2020-12-08 at 11 23 48 AM" src="https://user-images.githubusercontent.com/1641214/101510418-e9a07380-3947-11eb-808c-682787241aa3.png">

RaphaelAmadeu commented:
> Visually looks broken to me, can you take a look @RaphaelAmadeu ?
> 
> <img alt="Screen Shot 2020-12-08 at 11 23 48 AM" width="993" src="https://user-images.githubusercontent.com/1641214/101510418-e9a07380-3947-11eb-808c-682787241aa3.png">
> 
> Also, the props table isn't loading:
> 
> <img alt="Screen Shot 2020-12-08 at 11 25 45 AM" width="1241" src="https://user-images.githubusercontent.com/1641214/101510617-279d9780-3948-11eb-8583-510177a2fc93.png">

I've addressed these issues on my latest commit, thanks!

jeffchew commented:
Thanks @RaphaelAmadeu ! can you check the web components ci-checks? probably snapshots...

jeffchew reviewed (approved):
Thank you @RaphaelAmadeu !
